server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java

```diff
@@ -158,57 +151,60 @@ public HierarchyCircuitBreakerService(Settings settings, ClusterSettings cluster
             this::setRequestBreakerLimit);
         clusterSettings.addSettingsUpdateConsumer(ACCOUNTING_CIRCUIT_BREAKER_LIMIT_SETTING, ACCOUNTING_CIRCUIT_BREAKER_OVERHEAD_SETTING,
             this::setAccountingBreakerLimit);
+        clusterSettings.addAffixUpdateConsumer(CIRCUIT_BREAKER_LIMIT_SETTING, CIRCUIT_BREAKER_OVERHEAD_SETTING,
+            this::setCircuitBreakerLimit,
+            (s, t) -> {});
+    }
+
+    @Override
+    public void registerNewCircuitBreakers(List<CircuitBreaker> circuitBreakers) {
```

Comment:
For plugins to create their own circuit breaker instances (so that they can maintain a reference to them without the service), we still need to register circuit breakers outside of the constructor. 

Main difference here is that registration no longer overwrites previous circuit breakers and will throw if one is already registered with the same name.